app/plugin.go

```diff
@@ -122,20 +128,25 @@ func (a *App) SyncPluginsActiveState() {
 
 			// Activate plugin if enabled
 			if pluginEnabled {
```

Comment:
This can result in a race condition, where while a go-routine is attempting to deactivate a plugin, the second go-routine might be checking its state from `config` and end up re-activating it. I would recommend to just create two slices, one that contains a list of plugins to enable and the other one to deactivate, then you can parallelize those operations.


Race condition that might happen
```
go-routine-1 starts (line 103)
go-routine-2 starts (line 132)

go-routine-1 checks pluginEnabled state of plugin-1, starts deactivating plugin-1
go-routine-1 successfully deactivates plugin-1
go-routine-2 checks pluginEnabled state of plugin-1
go-routine-2 ends up reactivating plugin-1 
...
```

This might be mitigated if the `config.PluginStates` is a snapshot when checking states. Even if it is I would still recommend creates two slices to make it clear remove any interdependency.

Reply:
> This might be mitigated if the config.PluginStates is a snapshot when checking states.

@ali-farooq0 By this, did you mean to dereference the pointer to the config at the beginning of the function?

Reply:
@mickmister Nope, i meant it's already being done at the top `config := a.Config().PluginSettings`. But wasn't sure if it actually creates a snapshot or maintains a live copy that others can change underneath.

Reply:
Traversing the code, `Config()` returns a pointer and does nothing underneath creates a snapshot

Reply:
However, the function `a.UpdateConfig` does create a clone before editing, so there shouldn't ever be any code that is directly editing the config without being within the guard of `UpdateConfig`